util-core/src/main/scala/com/twitter/util/Time.scala

```diff
@@ -196,12 +198,22 @@ trait TimeLike[This <: TimeLike[This]] extends Ordered[This] { self: This =>
   def diff(that: This): Duration
 
   /**
+   * Rounds up to the nearest multiple of the given duration.  For example:
+   * 127.seconds.ceiling(1.minute) => 3.minutes.  Taking the ceiling of a
+   * Time object with duration greater than 1.hour can have unexpected
+   * results because of timezones.
+   */
+  // the impl might seem weird but consider the case where this is an exact multiple of the 
+  // increment.
+  def ceil(increment: Duration): This = (this - 1.nanoseconds).floor(increment) + increment
```

Comment:
sorry to be anal about this, but slow libs are often slow because of death by a thousand cuts.  `1.nanoseconds` adds an allocation, can we val this up?